fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/WhereFragment.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * (C) Copyright IBM Corp. 2019,2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.database.utils.query;
+
+/**
+ * An adapter which represents just a fragment of the where clause
+ * so we can build up just a portion of the predicate without having
+ * access to the whole clause
+ */
```

Comment:
What's the use case for this?  more a curiosity question.

Reply:
When building just a portion of a where clause, such as the expression we want to build for a date-range parameter. It's not a full where clause, because you can't add a group-by, order-by or having clause. That's why we use the BaseWhereAdapter to enforce that separation.